crates/bevy_ui/src/ui_node.rs

```diff
@@ -25,6 +27,18 @@ use smallvec::SmallVec;
 /// - [`Interaction`](crate::Interaction) to obtain the interaction state of this node
 #[derive(Component, Debug, Copy, Clone, PartialEq, Reflect)]
 #[reflect(Component, Default, Debug)]
+#[require(
+    Style,
+    BackgroundColor,
+    BorderColor,
```

Comment:
I'm not sure that we should require `BackgroundColor` and `BorderColor` for `Node`, it means that we can't distinguish between visable and non-visable nodes intended only to constrain the layout. 

Also extraction for `UiMaterial` nodes has a `Without<BackgroundColor>` query filter, so UI materials is broken atm with this PR. That can be fixed by just removing the query filter from `bevy_ui::render::ui_material_pipeline::extract_ui_material_nodes`  but maybe it goes against the intent of the API, I'm not sure.

Reply:
Yeah IMO we should remove this require.

Reply:
`BackgroundColor` and `BorderColor` are only accessed internally during extraction, so leaving them out shouldn't cause any problems. I ran some naive benchmarks and the cost of extracting all these extra uinodes can become significant if you have a very complicated layout.

Reply:
I think we should be conservative here given how close we are to release. Lets give this conversation its own space / issue where we can look at the benchmarks and discuss the implications of not having these everywhere.